app/controllers/sign_up_sheet_controller.rb

```diff
@@ -283,8 +283,152 @@ def set_priority
     end
 
     redirect_to action: 'list', assignment_id: params[:assignment_id]
+end
+
+
+  # This function works in conjunction with Add Team
+  def assign_topic # renders a view for a team to be added
+    @topic = SignUpTopic.find(params[:id]) # Identify the topic to which you want to add team
+    @assignment = Assignment.find(params[:assignment_id])
+  end
+
+  # This function works in conjunction with Remove Team
+  def remove_topic # renders a view for the team to be removed
+    @topic = SignUpTopic.find(params[:id])
+    @assignment = Assignment.find(params[:assignment_id]) # Identify the assignment
+  end
+
+  def remove_team # method for removing the team assigned to the topic
+    @topic = SignUpTopic.find(params[:id]) #Find all the details about the topic
+    @assignment = Assignment.find(params[:assignment_id]) # Find all the details about the assignment
+    @user = User.find_by name: params[:user][:name] # Find user
+
+    if @user.blank? # Check if the user exists
+      flash[:error] = "User does not exist"
+      redirect_to edit_assignment_path(params[:assignment_id]) + "#tabs-2"
+    else
+      @team = TeamsUser.find_by_sql("select T1.team_id from teams_users T1, teams T2 where T1.user_id = "+ @user.id.to_s+" AND T1.team_id = T2.id AND T2.parent_id = "+ @assignment.id.to_s) # find a team associated with this user having particular assignment and topic
```

Comment:
It is not a good idea to use "find_by_sql". 
Plus this functionality has already been implemented, see select T1.team_id from in TeamsUsers class.